### PR TITLE
Assign user roles

### PR DESCRIPTION
**Pull request by dallascrichmond**
## 🎯 Summary
- Added the ability to get a list of users based on their first name (could add more search fields to query) using getAzureIDIRUsers to start (Could add getIDIRUsers as well)
- Added the ability to assign a user a role by entering a role name followed by the user's GUID@idir (Some improvement could be made here, but I wanted to chat to you before I did anything else)

  
[<EDIT-ME>](https://apps.itsm.gov.bc.ca/jira/browse/<EDIT-ME>)

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: src/backend/src/modules/cssAPI/role/controller.ts [hunk 1]
```diff
@@ -41,6 +47,33 @@ export const getKCRole = errorWrapper(async (req: Request, res: Response) => {
   res.json(await getRole(role));
 });
 
+/**
+ * @method GET
+ * @param user - The user name to search.
+ * @route /cssAPI/getUser
```
Comment (BradyMitch) [inline src/backend/src/modules/cssAPI/role/controller.ts]:
```suggestion
 * @route /cssAPI/user/users
```

File: src/backend/src/modules/cssAPI/role/controller.ts [hunk 2]
```diff
@@ -41,6 +47,33 @@ export const getKCRole = errorWrapper(async (req: Request, res: Response) => {
...
+ * @protected Requires "playground-admin"
+ */
+export const getKCUser = errorWrapper(async (req: Request, res: Response) => {
+  if (DEBUG)
+    console.info("DEBUG: getKCUser controller in modules/cssAPI called.");
+
+  const user: IDIRUserQuery = {
+    firstName: req.query.user as string,
```
Comment (BradyMitch) [inline src/backend/src/modules/cssAPI/role/controller.ts]:
```suggestion
    firstName: req.query.firstName as string,
```

File: src/frontend/src/pages/CSSAPIPage.tsx [hunk 3]
```diff
@@ -283,6 +286,80 @@ const CSSAPIPage = () => {
...
+                  ></input>
+                  <Button
+                    size="s"
+                    onClick={() => {
+                      if (getUserInput !== "")
+                        callAPI(
+                          "/cssAPI/getKCUser",
+                          "GET",
+                          `?user=${getUserInput}`
+                        );
```
Comment (NoorChasib) [inline src/frontend/src/pages/CSSAPIPage.tsx]: I think we should abstract the onclicks out as a function and pass that back in here

File: src/frontend/src/pages/CSSAPIPage.tsx [hunk 4]
```diff
@@ -283,6 +286,80 @@ const CSSAPIPage = () => {
...
+                    onChange={(e) => setUserIDIRInput(e.target.value)}
+                  ></input>
+                  <Button
+                    size="s"
+                    onClick={() => {
+                      if (userIDIRInput !== "")
+                        callAPI(
+                          `/cssAPI/assignUserRole?role=${assignUserRoleInput}&user=${userIDIRInput}`,
+                          "POST"
+                        );
```
Comment (NoorChasib) [inline src/frontend/src/pages/CSSAPIPage.tsx]: same here

File: src/backend/src/modules/cssAPI/role/controller.ts [hunk 5]
```diff
@@ -62,6 +95,31 @@ export const createKCRole = errorWrapper(
   }
 );
 
+
+/**
+ * @method POST
+ * @param role - The role name to assign.
+ * @param user - The user's GUID
+ * @route /cssAPI/assignUserRoles
```
Comment (BradyMitch) [inline src/backend/src/modules/cssAPI/role/controller.ts]:
```suggestion
 * @route /cssAPI/role/assign/:guid
```

**Pull request closed without merging**